templates/github/.ci/scripts/check_release.py.j2

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+{% include 'header.j2' %}
+
+import argparse
+import re
+import os
+import yaml
+from tempfile import TemporaryDirectory
+from packaging.version import Version
+from git import Repo
+
+UPSTREAM_REMOTE = "https://github.com/{{ github_org }}/{{ plugin_name }}.git"
+DEFAULT_BRANCH = "{{ plugin_default_branch }}"
+RELEASE_BRANCH_REGEX = r"^([0-9]+)\.([0-9]+)$"
+Y_CHANGELOG_EXTS = [".feature", ".removal", ".deprecation"]
+Z_CHANGELOG_EXTS = [".bugfix", ".doc", ".misc"]
+
+
+def main():
+    """Check which branches need a release."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--branches",
+        default="supported",
+        help="A comma separated list of branches to check for releases. Can also use keyword: "
+        "'supported'. Defaults to 'supported', see `ci_update_branches` in "
+        "`plugin_template.yml`.",
+    )
+    opts = parser.parse_args()
+
+    with TemporaryDirectory() as d:
+        # Clone from upstream to ensure we have updated branches & main
+        repo = Repo.clone_from(UPSTREAM_REMOTE, d, filter="blob:none")
+        heads = [h.split("/")[-1] for h in repo.git.ls_remote("--heads").split("\n")]
+        branches = [h for h in heads if re.search(RELEASE_BRANCH_REGEX, h)]
+        branches.sort(key=lambda ver: Version(ver))
+
+        versions = opts.branches
+        check_y_release = False
+        if versions == "supported":
+            with open(f"{d}/template_config.yml", mode="r") as f:
+                tc = yaml.safe_load(f)
+                versions = tc["ci_update_branches"]
+            check_y_release = True
+        else:
+            splitted = versions.split(",")
+            versions = [v for v in splitted if re.match(RELEASE_BRANCH_REGEX, v) and v in branches]
+            if len(versions) != len(splitted):
+                print("Supplied branches contains non-existent branches!")
+                exit(1)
+
+        print(
+            f"Checking for releases on branches: {versions}",
+            f"and {DEFAULT_BRANCH}" if check_y_release else "",
```

Comment:
Can one check for the y-release without all the z-releases?

Is it even worth adding all this logic? We could just be opinionated and just always check all release branches and the default branch.